src/test/librbd/test_mirroring.cc

```diff
@@ -1137,9 +1137,10 @@ TEST_F(TestMirroring, Snapshot)
   ASSERT_EQ(0, image.mirror_image_create_snapshot(&snap_id));
   vector<librbd::snap_info_t> snaps;
   ASSERT_EQ(0, image.snap_list(snaps));
-  ASSERT_EQ(2U, snaps.size());
-  ASSERT_EQ(snaps[1].id, snap_id);
+  ASSERT_EQ(1U, snaps.size());
```

Comment:
The reasoning behind this change is that the initial snapshot is created when there is no peers. I can add a comment in the code if you want.

Reply:
What do you mean by "treating them specially"? Technically this commit add a special case for them and a unit test so I'm a bit confused. But yes indeed at some point in my fix proposal I did check for this this case only on the snapshot "naturally" selected for removal. But after all a mirror snapshot with no peers is not really useful, so it could make sense to just delete them right away.

Reply:
Additionally to the case I described above, there is another case when "no peer" snapshot may be created. This is if the mirroring pool is not configured properly yet (no peer is added to the pool) and a mirror image is created (mirroring is enabled). Exactly like in the test case here.

When mirroring is enabled for an image, a primary snapshot is created. This is needed to distinguish primary image from non-primary. Probably we should just not allow to enable mirroring (return -EINVAL) when the pool does not have any peer, like we do for `mirror_image_create_snapshot` API call (see this test above). But right now I am not ready to tell if there is no a valid case that will be broken by this change.

So this initial snapshot (created on enabling mirroring) is now deleted when a new valid snapshot is created.

Reply:
By treating them specially I meant prioritizing their removal.  It would be a larger change though and probably not worth it.  @MrFreezeex Let's add a comment, like you initially offered to.

Reply:
Sorry, Github was showing my previous "treating them specially" comment in two different places for me.  I removed it in the other place thinking I somehow commented twice, but it also got removed from this thread.